lib/system/web/src/animationframeloop.rs

```diff
@@ -0,0 +1,60 @@
+use super::request_animation_frame;
+use wasm_bindgen::prelude::Closure;
+use std::rc::Rc;
+use std::cell::RefCell;
+
+
+// ============================
+// === Animation Frame Data ===
+// ============================
+
+struct AnimationFrameData {
+    run : bool
+}
+
+pub struct AnimationFrameLoop {
+    forget : bool,
+    data   : Rc<RefCell<AnimationFrameData>>
+}
```

Comment:
You've mentioned that you want to merge this animation frame loop with or other one. Is this merge going to happen in another PR?

Reply:
This will happen when we get rid of our duplicates